core/interfaces/i_icon.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Coordinate} from '../utils/coordinate.js';
+import {Size} from '../utils/size.js';
+
+
+export interface IIcon {
+  /**
+   * @return the string representing the type of the icon.
+   * E.g. 'comment', 'warning', etc. This string should also be used when
+   * registering the icon class.
+   */
+  getType(): string;
+
+  /**
+   * Creates the SVG elements for the icon that will live on the block.
+   *
+   * @param pointerdownListener The listener to be attached to the root svg
```

Comment:
Suggestion:

```
@param pointerdownListener An event listener that must be attached to the
root SVG element by the implementation of `initView`. Used by Blockly's
gesture system to properly handle clicks and drags.
```

This includes "must be attached" rather than "to be attached", so that it doesn't look automatic.